Creating CONTRIBUTING.md

kyang4 opened the pull request:
## Description
<!-- Provide a brief description of the changes in this PR -->

First draft of CONTRIBUTING.md, including commit formats. 

## Changes Made
<!-- List the key changes made in this PR -->

## How to Test

## Notes
<!-- Any further notes regarding changes -->

kyang4 committed:
Creating CONTRIBUTING.md

neelneelneel committed:
Update CONTRIBUTING.md

neelneelneel commented:
This is good to go from my perspective.

@johbaxter and @EJV20 - can you validate that the commit messages match the lint rules?

EJV20 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,148 @@
+* `(legacy)`: The "old" UI, the BI tool, and terminal
+* `(ui)`: Base component library for the UI
+* `(renderer)`: Library used for Drag-and-Drop App Builder
+* `(sdk)`: Tools to help developers tap into SEMOSS enviornment
```
This is not accurate for projects other than the UI. For most projects, We allow it to be open ended since most code bases do not fit nicely into 5 sections.

EJV20 reviewed (changes requested):
Perfect for semoss-ui repo. Other repos have different commit syntax around the commit scope.

EJV20 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,148 @@
+
+For example ticket number 10, with type test, and scope api with the commit message being 'added api tests', your commit would be:
+
+`[#10] test(api): added api tests`
```
This also, would not pass according to rules laid out below

EJV20 replied:
Commit description could be a little more flushed out in the example. I know I probably have the same lingo in the Semoss readme.md

kyang4 committed:
Update CONTRIBUTING.md

Replaced scope section with blurb, updated language on past tense to present imperative tense, filled in missing Type descriptions, other small updates.

kyang4 commented:
just made a new commit to address the changes requested
